parantheses/parantheses.go

```diff
@@ -0,0 +1,53 @@
+package parantheses
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"strconv"
+)
+
+func IsBalanced(s string) bool {
+	stack := []rune{}
+	mapping := map[rune]rune{'(': ')', '[': ']', '{': '}'}
+
+	for _, c := range s {
+		if c == '(' || c == '[' || c == '{' {
+			stack = append(stack, c)
+		} else if c == ')' || c == ']' || c == '}' {
+			if len(stack) == 0 {
+				return false
+			}
+
+			last := stack[len(stack)-1]
+			stack = stack[:len(stack)-1]
+
+			if mapping[last] != c {
+				return false
+			}
+		}
+	}
+
+	return len(stack) == 0
+}
+
+func GenerateParantheses(w http.ResponseWriter, r *http.Request) {
```

Comment:
It's quite important to separate "business logic" from the way it's being used. And in general it is important to separate and abstract things/system components from each other. That allows us to easily modify our system to achieve what business needs. 

In this case, generating parentheses has nothing to do with HTTP, right? These system components don't need to/have to know about each other. By writing the `GenerateParantheses()` as an `http.Hander` you have added a dependency. Your bracket generation _depends_ on the fact that it is used as a `Handler`. Dependencies can be fast to implement in the moment, but they are very limiting in the future.

Tomorrow your boss will tell you: "hey, I really like your bracket generation web service, can you make the same functionality as a CLI? I'd really like to run it as a command in a terminal: `./parentheses-gen -n 100`". 
Is it easy for you to add such option? 

A week later a team member will want to use this `func` directly, but it requires him to create http.Request, set params and so on. 
Is it easy for your coworker to do that?

Tell me here, what do you think is the best way to fix this?

Reply:
and `parentheses.GenerateParentheses()`. You already have `parentheses` in the package name, don't repeat it

Reply:
shall we separat the GenerateParentheses() function from the HTTP handle

Reply:
yes, please do that